flink-streaming-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractFetcher.java

```diff
@@ -207,15 +207,14 @@ public void restoreOffsets(HashMap<KafkaTopicPartition, Long> snapshotState) {
 	// ------------------------------------------------------------------------
 
 	/**
-	 * 
 	 * <p>Implementation Note: This method is kept brief to be JIT inlining friendly.
 	 * That makes the fast path efficient, the extended paths are called as separate methods.
-	 * 
 	 * @param record The record to emit
 	 * @param partitionState The state of the Kafka partition from which the record was fetched
-	 * @param offset The offset from which the record was fetched
+	 * @param offset The offset of the record
+	 * @param kafkaRecord The original Kafka record
 	 */
-	protected final void emitRecord(T record, KafkaTopicPartitionState<KPH> partitionState, long offset) {
+	protected <R> void emitRecord(T record, KafkaTopicPartitionState<KPH> partitionState, long offset, R kafkaRecord) throws Exception {
```

Comment:
Is there a reason we need to have an extra `kafkaRecord`? It doesn't seem to be used in the function.